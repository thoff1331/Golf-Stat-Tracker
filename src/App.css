.sign-out {
    margin-left: 90%;
    position: absolute;
    top: 0%;
    padding: 1rem;
    text-decoration: none;
    margin-top: 10px;
    display: inline-block;
    background-color: #ffdf00;
    color: #000000;
    border-radius: 4px;
}

.sign-out:hover {
    color: white;
    background-color: #42545c;
    text-decoration: none;
  }
  .sign-out .sign-out.active {
    background-color: black;
    text-decoration: none;
  }