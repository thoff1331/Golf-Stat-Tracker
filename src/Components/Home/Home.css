.home-container {
    display:  flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
}

.image-scroller {
font-size: 60px;
color: #ffdf00; 
margin: 24px;   
}

img {
    height: 400px;
    width: 400px;
    border: 3px solid #ffdf00;
}