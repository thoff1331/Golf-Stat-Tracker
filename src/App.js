import "@aws-amplify/ui-react/styles.css";
import Home from './Components/Home/Home'
import './App.css'


export default function App({ signOut }) {
  return (
   <Home />
  );
}

