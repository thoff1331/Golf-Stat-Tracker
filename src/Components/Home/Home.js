import * as React from 'react';
import './Home.css'
import {useState} from 'react'
import * as imgs from '../../constants/HomeConstants.json'

function Home() {
    const [imgPage,setImgPage] = useState(0)
    const imageIndex = Array.from(imgs)  
    const numOfImages = imageIndex.length
    function nextImg() {
     setImgPage( imgPage === numOfImages - 1 ? 0 : imgPage + 1)   
    }
    function prevImg() {
        setImgPage( imgPage === 0  ? 0 : imgPage - 1)   
       }

  return (
    <div className='home-container'>
<button onClick={() => prevImg()} className='image-scroller' data-testid="button">{"<"}  </button>
<img src={imgs[imgPage].src}  alt='golf-news' />
<button  onClick={() => nextImg()}  className='image-scroller'> {">"}</button>
</div>
  );
}

export default Home;