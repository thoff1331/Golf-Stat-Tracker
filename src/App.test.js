import React from 'react'
import {screen,render, cleanup} from '@testing-library/react'
import App from './App'
import { assert } from 'chai';
import {nextImg,prev,Img,Home} from "./Components/Home/Home"

 afterEach(cleanup)
 
 it('should take a snapshot', () => {
    const { asFragment } = render(<App />)
    
    expect(asFragment(<App />)).toMatchSnapshot()
   });


describe('tests for Home Component', function() {
   render(<button onClick={() => prevImg()} className='image-scroller'>{"<"}  </button>); 
   const button = screen.queryByTestId("button"); 
   it('returns the first image', function ()  {
     assert.equal(nextImg,undefined)  
   } )  
   test("Button Rendering", () => {
      expect(button).not.toBeInTheDocument(); 
  })
 })


   