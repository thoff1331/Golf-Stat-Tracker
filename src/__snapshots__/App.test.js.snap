// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-container"
  >
    <button
      class="image-scroller"
      data-testid="button"
    >
      &lt;  
    </button>
    <img
      alt="golf-news"
      src="https://igp.brightspotcdn.com/dims4/default/09fd6d9/2147483647/strip/true/crop/1024x498+0+87/resize/1926x936!/quality/90/?url=http%3A%2F%2Findigogolf-brightspot.s3.amazonaws.com%2Fclubs%2Fb0%2F23%2F28e764b24aefa19658bebc5d3b4f%2Fdubsdread.jpg"
    />
    <button
      class="image-scroller"
    >
       &gt;
    </button>
  </div>
</DocumentFragment>
`;
